tritium-metrics/src/main/java/com/palantir/tritium/metrics/TaggedMetricsScheduledExecutorService.java

```diff
@@ -35,52 +34,65 @@
     private final Counter scheduledOverrun;
-    private final Histogram scheduledPercentOfPeriod;
 
     TaggedMetricsScheduledExecutorService(ScheduledExecutorService delegate, ExecutorMetrics metrics, String name) {
         this.delegate = delegate;
         this.name = name;
 
         this.submitted = metrics.submitted(name);
         this.running = metrics.running(name);
-        this.completed = metrics.completed(name);
         this.duration = metrics.duration(name);
 
-        this.scheduledOnce = metrics.scheduledOnce(name);
-        this.scheduledRepetitively = metrics.scheduledRepetitively(name);
         this.scheduledOverrun = metrics.scheduledOverrun(name);
-        this.scheduledPercentOfPeriod = metrics.scheduledPercentOfPeriod(name);
     }
 
     @Override
     public ScheduledFuture<?> schedule(Runnable task, long delay, TimeUnit unit) {
-        scheduledOnce.mark();
-        return delegate.schedule(new TaggedMetricsRunnable(task), delay, unit);
+        ScheduledFuture<?> future = delegate.schedule(new TaggedMetricsRunnable(task), delay, unit);
+        // RejectedExecutionException should prevent 'submitted' from being incremented.
+        // This means a wrapped same-thread executor will produce delayed 'submitted' values,
+        // however the results will work as expected for the more common cases in which
+        // either a queue is full, or the delegate has shut down.
+        submitted.mark();
+        return future;
     }
 
     @Override
     public <V> ScheduledFuture<V> schedule(Callable<V> callable, long delay, TimeUnit unit) {
-        scheduledOnce.mark();
-        return delegate.schedule(new TaggedMetricsCallable<>(callable), delay, unit);
+        ScheduledFuture<V> future = delegate.schedule(new TaggedMetricsCallable<>(callable), delay, unit);
+        // RejectedExecutionException should prevent 'submitted' from being incremented.
+        // This means a wrapped same-thread executor will produce delayed 'submitted' values,
+        // however the results will work as expected for the more common cases in which
+        // either a queue is full, or the delegate has shut down.
+        submitted.mark();
+        return future;
     }
 
     @Override
     public ScheduledFuture<?> scheduleAtFixedRate(Runnable task, long initialDelay, long period, TimeUnit unit) {
-        scheduledRepetitively.mark();
-        return delegate.scheduleAtFixedRate(
+        ScheduledFuture<?> future = delegate.scheduleAtFixedRate(
                 new TaggedMetricsScheduledRunnable(task, period, unit), initialDelay, period, unit);
+        // RejectedExecutionException should prevent 'submitted' from being incremented.
+        // This means a wrapped same-thread executor will produce delayed 'submitted' values,
+        // however the results will work as expected for the more common cases in which
+        // either a queue is full, or the delegate has shut down.
+        submitted.mark();
```

Comment:
It's really confusing to use `submitted` for repetitive tasks. I typically view `submitted` as "how many tasks were submitted to the executor" and not "how many times was a executor method called".

Reply:
In this case a task is being submitted to the executor. It's not quite the same as a one-shot task in that it may be executed 0-N times rather than 0-1.
I could see an argument for excluding the metric entirely on recurring tasks, but I don't have a strong preference.

Reply:
Yeah I'd probably just exclude it entirely. I think the metric is significantly less useful when it's combined like this. I generally use `submitted` as a proxy for "how much work is being requested" and that doesn't really work when a single submitted task will be executed N times.

Reply:
Ya that's fair, at that point we might as well remove the submitted metric entirely from the scheduled executor because there's tricky interplay between recurring tasks and `execute`/`submit` which invalidates most of our assumptions anyhow.